dhis-2/dhis-services/dhis-service-analytics/src/main/java/org/hisp/dhis/analytics/resolver/DataElementGroupResolver.java

```diff
@@ -0,0 +1,102 @@
+/*
+ * Copyright (c) 2004-2021, University of Oslo
+ * All rights reserved.
+ *
+ * Redistribution and use in source and binary forms, with or without
+ * modification, are permitted provided that the following conditions are met:
+ * Redistributions of source code must retain the above copyright notice, this
+ * list of conditions and the following disclaimer.
+ *
+ * Redistributions in binary form must reproduce the above copyright notice,
+ * this list of conditions and the following disclaimer in the documentation
+ * and/or other materials provided with the distribution.
+ * Neither the name of the HISP project nor the names of its contributors may
+ * be used to endorse or promote products derived from this software without
+ * specific prior written permission.
+ *
+ * THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS" AND
+ * ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE IMPLIED
+ * WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE ARE
+ * DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT OWNER OR CONTRIBUTORS BE LIABLE FOR
+ * ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES
+ * (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES;
+ * LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON
+ * ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT
+ * (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE OF THIS
+ * SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+ */
+package org.hisp.dhis.analytics.resolver;
+
+import static com.google.common.base.Preconditions.checkNotNull;
+import static org.hisp.dhis.expression.ParseType.INDICATOR_EXPRESSION;
+
+import java.util.List;
+import java.util.Set;
+import java.util.stream.Collectors;
+
+import org.hisp.dhis.common.DimensionalItemId;
+import org.hisp.dhis.dataelement.DataElementGroup;
+import org.hisp.dhis.dataelement.DataElementGroupStore;
+import org.hisp.dhis.expression.ExpressionService;
+import org.springframework.stereotype.Service;
+
+import com.google.common.base.Joiner;
+
+/**
+ * @author Dusan Bernat
+ */
+
+@Service( "org.hisp.dhis.analytics.resolver.DataElementGroupResolver" )
+public class DataElementGroupResolver implements ExpressionResolver
+{
+    private final ExpressionService expressionService;
+
+    private final DataElementGroupStore dataElementGroupStore;
+
+    private static final String LEFT_BRACKET = "(";
+
+    private static final String RIGHT_BRACKET = ")";
+
+    private static final String DATA_ELEMENT_GROUP_PREFIX = "deGroup:";
+
+    private static final String EMPTY_STRING = "";
+
+    public DataElementGroupResolver( ExpressionService expressionService,
```

Comment:
This boilerplate constructor (and others like it in other classes) can be avoided by annotating the class with `@AllArgsConstructor` from
`import lombok.AllArgsConstructor;`
(see elsewhere in the project)

Reply:
done in all resolvers